ebu_tt_live/ui/user_input_producer/uip.js

```diff
@@ -0,0 +1,570 @@
+         $(document).ready(function(){
+
+            var sequence_numbers = {};
+            var all_sequences = {};
+            var interval_send = null;
+            var scheduled_send_media_clock_offset = null;
+            var interval_running_clock = null;
+
+            if (typeof(Storage) === "undefined") {
+               window.alert("You are using an old browser that does not allow to save data between session. Do not reload this page or you will lose your defined sequences.");
+            }
+
+
+            // Get data from local storage if it exists (we retain some data, like existing sequences and sequence numbers
+            // on page reload. Also works when the browser is closed and reopened.
+            if (localStorage.sequence_selector) {
+               $("#sequence-selector").html(localStorage.sequence_selector);
+            }
+            if (localStorage.all_sequences) {
+               all_sequences = JSON.parse(localStorage.all_sequences);
+            }
+            if (localStorage.sequence_numbers) {
+               sequence_numbers = JSON.parse(localStorage.sequence_numbers);
+            }
+
+
+            /********************************************************** Options to show or not depending on other options and general purpose functions****************************/
+
+            function stopResetRunningClock() {
+                  $("#running-media-clock").html("");
+                  if (interval_running_clock != null) {
+                     clearInterval(interval_running_clock);
+                     interval_running_clock = null;
+                  }
+            }
+
+            function handleTimeBaseDependingOptions() {
+               var selected_time_base = $("#time-base-selector").val();
+               if (selected_time_base == "clock") {
+                  $("#clock-mode-span").show();
+                  $("#smpte-attributes-span").hide();
+               } else if (selected_time_base == "smpte") {
+                  $("#clock-mode-span").hide();
+                  $("#smpte-attributes-span").show();
+               } else {
+                  $("#clock-mode-span").hide();
+                  $("#smpte-attributes-span").hide();
+               }
+            }
+
+            handleTimeBaseDependingOptions();
```

Comment:
Personally i'd prefer if the calls for functions are done after they have all been defined. They get lost when they are just under the function.